src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,22 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} spoiler Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler) {
+    if (spoiler === true) {
```

Comment:
You should do the reverse validation here to avoid ending up with `SPOILER_SPOILER_name.png` or something